### ISSUE TITLE
Crash on import of large DXF file

### ISSUE THREAD

**Issue by blsemo** [SpaceGroupUCL/depthmapX#70]
User report of crash when importing a very large DXF file on Mac os.

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x00000000000000d0
Exception Note:        EXC_CORPSE_NOTIFY

Thread 8 Crashed:: RenderThread
0   net.orangevertex.artk6.depthmapX    0x000000010a8c80aa DxfParser::openEntities(std::__1::basic_istream<char, std::__1::char_traits<char> >&, DxfToken&, DxfBlock*) + 3146
1   net.orangevertex.artk6.depthmapX    0x000000010a8c5ded DxfParser::open(std::__1::basic_istream<char, std::__1::char_traits<char> >&) + 1149
2   net.orangevertex.artk6.depthmapX    0x000000010a8c595f operator>>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, DxfParser&) + 143
3   net.orangevertex.artk6.depthmapX    0x000000010a81073a MetaGraph::loadDxf(std::__1::basic_istream<char, std::__1::char_traits<char> >&, Communicator*) + 378
4   net.orangevertex.artk6.depthmapX    0x000000010a80fffd MetaGraph::loadLineData(Communicator*, int) + 813
5   net.orangevertex.artk6.depthmapX    0x000000010a76defd RenderThread::run() + 125
6   org.qt-project.QtCore             0x000000010b4ec85a 0x10b4c2000 + 174170
7   libsystem_pthread.dylib           0x00007fffa6bc393b _pthread_body + 180
8   libsystem_pthread.dylib           0x00007fffa6bc3887 _pthread_start + 286
9   libsystem_pthread.dylib           0x00007fffa6bc308d thread_start + 13

**Comment by yuan7220** [comment]
I used 64bit windows DepthMapX0.50 which crashed at 85% or 86%  when loading in a 300mb dxf file. Any suggestions?

**Comment by orange-vertex** [comment]
Unfortunately "300mb dxf file" is not sufficient for us to understand what the problem is. Can you send us the file or give us an idea of the number of objects in that file i.e. how many lines, polylines, circles, arcs, blocks and splines are in there?

**Comment by yuan7220** [comment]
Thanks for checking. I just tried another smaller file 115MB, but still crashed at about 85-87% when importing into DepthMapX. The DXF can be downloaded from https://www.dropbox.com/s/guhgassez0uqn32/nctcentr.DXF?dl=0 

Using  Map > import, the program will load the DXF file until 85-87% as shown on the screen.  Then an error message shows that a problem caused the program to stop working correctly. Please close the program. 

The DXF is created using ArcGIS to create buffers of street widths from a road network for the city of Dallas. I can open the DXF file in AutoCad without problems. I attempt to use the road buffer polygon (only one for the entire city) to create Axial lines and derive Space Syntax measurement. 

thanks very much for your help
May

**Comment by orange-vertex** [comment]
Thank you for providing this file May! Indeed I can replicate the problem though only on Windows (Mac loads your file just fine). I don't know what the problem is exactly, but it's fixed in the newest versions (we've worked a bit on the DXF importing) soon to be released. I suspect its the fact that everything is in a block, because as soon as I exploded it it loaded even on 0.50 x64.

So you have three options:
1. Explode the block in autocad and load it with any version
2. Work with a beta unreleased version from [here](https://github.com/SpaceGroupUCL/depthmapX/tree/map_opengl/dist/beta/map_opengl) (you might also need [this](https://www.microsoft.com/en-us/download/details.aspx?id=48145) for windows)
3. Switch to a mac
